tests/test_reactive.py

```diff
@@ -598,3 +598,68 @@ def compose(self) -> ComposeResult:
     app = MyApp()
     async with app.run_test():
         assert app.query_one(MyWidget).foo == "foobar"
+
+
+async def test_no_duplicate_external_watchers() -> None:
+    """Make sure we skip duplicated watchers."""
+
+    counter = 0
+
+    class Holder(Widget):
+        attr = var(None)
+
+    class MyApp(App[None]):
+        def __init__(self) -> None:
+            super().__init__()
+            self.holder = Holder()
+
+        def on_mount(self) -> None:
+            self.watch(self.holder, "attr", self.callback)
```

Comment:
The test name/docstring says it's checking that we skip duplicate watchers, but the test no longer involves a duplicate watcher. It seems to just be checking the basic usage of `watch`, which is probably covered by another test.

Reply:
As for the first two paragraphs, I think I deleted the second callback when I was debugging the bug I mentioned in a previous commit and then forgot to add it again, thanks for spotting that...

As for the third paragraph... Do you mean something like the test `test_external_watch_init_does_not_propagate_to_externals` added in 8b349e3?